queries/nix/injections.scm

```diff
@@ -15,7 +15,7 @@
     (string_expression (string_fragment) @bash)
     (indented_string_expression (string_fragment) @bash)
   ]
-  (#match? @_path "(^\\w*Phase|(pre|post)\\w*|(.*\\.)?\\w*([sS]cript|[hH]ook)|(.*\\.)?startup)$"))
+  (#match? @_path "(^\\w+(Phase|Hook)|(pre|post)[A-Z]\\w+|script)$"))
```

Comment:
to me your version also looks cleaner. I still want to know what the SEGFAULT is?